crates/scheduler/src/call/platform/update.rs

```diff
@@ -57,9 +57,8 @@ fn metadata<B: Board>(mut call: SchedulerCall<B, api::metadata::Sig>) {
                 memory.get_mut(ptr, len)?.copy_from_slice(&metadata);
                 memory.get_mut(*ptr_ptr, 4)?.copy_from_slice(&ptr.to_le_bytes());
                 memory.get_mut(*len_ptr, 4)?.copy_from_slice(&len.to_le_bytes());
-                Ok(())
             }
-            Err(error) => Err(error),
+            Err(error) => Err(error)?,
```

Comment:
Same here, we can get rid of this branch, and actually the whole match by having:

```rust
let metadata = board::platform::Update::<B>::metadata()?;
```